### PR TITLE
Feature/csb 8237

### PR DESCRIPTION
**Pull request by KHodderDVSA**
Increase test coverage. Currently at 84%, with only the handler outstanding for which we don't yet have a satisfactory testing solution.
image

### CODE REVIEW

File: tests/unit/testNumberService.unitTest.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                    id: "W99",
+                    certLetter: "Z",
+                    sequenceNumber: "998"
+                };
+                const newTestNumber = testNumberService.createNextTestNumberObject(lastTestNumber);
+                // next number should be W99Z999 = [23 + 9 + 27 + 26 + 27 + 9 + 9] = 130 => 30;
+                expect("30").toEqual(newTestNumber.testNumber.substring(7, 9));
+            });
+
+            it("Should have correct CheckSum numbers (checksum < 10)", () => {
```
Comment (arupdvsa) [inline tests/unit/testNumberService.unitTest.ts]: missing async

File: tests/unit/testNumberService.unitTest.ts [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                    id: "X99",
+                    certLetter: "Q",
+                    sequenceNumber: "919"
+                };
+                const newTestNumber = testNumberService.createNextTestNumberObject(lastTestNumber);
+                // next number should be X99Q920 = [24 + 9 + 27 + 17 + 9 + 6 + 0] = 92;
+                expect("92").toEqual(newTestNumber.testNumber.substring(7, 9));
+            });
+
+            it("Should have correct CheckSum numbers (checksum > 100)", () => {
```
Comment (arupdvsa) [inline tests/unit/testNumberService.unitTest.ts]: missing async

File: tests/unit/testNumberService.unitTest.ts [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                expectedNextTestNumber = {
+                    testNumber: "X01A00129",
+                    id: "X01",
+                    certLetter: "A",
+                    sequenceNumber: "001"
+                };
+                expect(testNumberService.createNextTestNumberObject(lastTestNumber)).toEqual(expectedNextTestNumber);
+            });
+
+            it("Should have correct CheckSum numbers (10 < checksum < 100)", () => {
```
Comment (arupdvsa) [inline tests/unit/testNumberService.unitTest.ts]: missing async

File: tests/unit/testNumberService.unitTest.ts [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+describe("TestNumberService", () => {
+    const testNumberService = new TestNumberService(new DynamoDBService());
+    beforeEach(() => {
+        jest.restoreAllMocks();
+        jest.resetModules();
+    });
+    describe("createNextTestNumberObject", () => {
+        context("corner cases", () => {
+            it("should return proper nextTestNumber", () => {
```
Comment (arupdvsa) [inline tests/unit/testNumberService.unitTest.ts]: missing async

File: tests/unit/configuration.unitTest.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        it("should return an error for missing functions from getFunctions", () => {
+            const config = new Configuration("../../tests/resources/badConfig.yml");
+            try {
+                config.getFunctions();
+            } catch (e) {
+                expect(e.message).toEqual("Functions were not defined in the config file.");
+            }
+        });
+
+        it("should return an error for missing DB Config from getDynamoDBConfig", () => {
```
Comment (arupdvsa) [inline tests/unit/configuration.unitTest.ts]: missing async

File: tests/unit/configuration.unitTest.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            expect(functions.length).toEqual(1);
+            expect(functions[0].name).toEqual("generateTestNumber");
+
+            const DBConfig = configService.getDynamoDBConfig();
+            expect(DBConfig).toEqual(configService.getConfig().dynamodb.remote);
+        });
+    });
+
+    context("with bad config file", () => {
+        it("should return an error for missing functions from getFunctions", () => {
```
Comment (arupdvsa) [inline tests/unit/configuration.unitTest.ts]: missing async

File: tests/unit/configuration.unitTest.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            const configService = Configuration.getInstance();
+            const functions = configService.getFunctions();
+            expect(functions.length).toEqual(1);
+            expect(functions[0].name).toEqual("generateTestNumber");
+
+            const DBConfig = configService.getDynamoDBConfig();
+            expect(DBConfig).toEqual(configService.getConfig().dynamodb["local-global"]);
+        });
+
+        it("should return remote versions of the config by default", () => {
```
Comment (arupdvsa) [inline tests/unit/configuration.unitTest.ts]: missing async

File: tests/unit/configuration.unitTest.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+            const functions = configService.getFunctions();
+            expect(functions.length).toEqual(1);
+            expect(functions[0].name).toEqual("generateTestNumber");
+
+
+            const DBConfig = configService.getDynamoDBConfig();
+            expect(DBConfig).toEqual(configService.getConfig().dynamodb.local);
+        });
+
+        it("should return local-global versions of the config if specified", () => {
```
Comment (arupdvsa) [inline tests/unit/configuration.unitTest.ts]: missing async

File: tests/unit/configuration.unitTest.ts [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    let branch: string | undefined = "";
+    beforeAll(() => {
+        branch = process.env.BRANCH;
+    });
+    afterAll(() => {
+        process.env.BRANCH = branch;
+    });
+
+    context("with good config file", () => {
+        it("should return local versions of the config if specified", () => {
```
Comment (arupdvsa) [inline tests/unit/configuration.unitTest.ts]: missing async

File: package.json [hunk 10]
```diff
@@ -8,14 +8,14 @@
     "build": "node_modules/typescript/bin/tsc --rootDir ./ --outDir .build --sourceMap false && npm run build:copy",
     "build:dev": "node_modules/typescript/bin/tsc && npm run build:copy",
     "build:copy": "find src -type f \\( -name \"*.yml\" -or -name \"*.json\" \\) | cpio -pdm .build && find tests -type f \\( -name \"*.yml\" -or -name \"*.json\" \\)  | cpio -pdm .build",
-    "test": "node_modules/.bin/mocha .build/tests/**/*.unitTest.js",
-    "test-i": "node_modules/.bin/mocha .build/tests/**/*.intTest.js",
+    "test": "jest --testMatch=\"**/*.unitTest.ts\"",
+    "test-i": "jest --testMatch=\"**/*.intTest.ts\"",
```
Comment (arupdvsa) [inline package.json]: include --runInBand

File: package.json [hunk 11]
```diff
@@ -8,14 +8,14 @@
...
     "tslint": "tslint src/**/*.ts tests/**/*.ts",
     "prepush": "BRANCH=local npm test && npm run tslint",
     "predeploy": "npm install && npm run prepush",
     "security-checks": "git secrets --scan && git log -p | scanrepo",
     "sonar-scanner": "npm run tslint && npm run coverage && sonar-scanner",
-    "coverage": "nyc mocha tests/unit/*.unitTest.ts && nyc report --reporter=lcov",
+    "coverage": "jest --testMatch=\"**/*.*Test.ts\" --coverage",
```
Comment (arupdvsa) [inline package.json]: include --runInBand

File: tests/unit/testNumberService.unitTest.ts (hunk 4 above)
Comment (KHodderDVSA) [reply tests/unit/testNumberService.unitTest.ts]: No I didn't - createNextTestNumberObject isn't asynchronous, so not required.
